source/includes/connection-snippets/x509-snippets/go-connection.go

```diff
@@ -0,0 +1,33 @@
+// begin x509 connection
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Trainer struct {
+    Name string
+    Age  int
+    City string
+}
+
+func main() {
+	ctx := context.TODO()
+	caFilePath := "/etc/certs/mongodb/ca.pem"
+	certificateKeyFilePath := "/etc/certs/mongodb/client.pem"
+
+	uri := "mongodb+srv://<your-x509-client-subject>@<cluster-url>/?tlsCAFile=%s&tlsCertificateKeyFile=%s&authMechanism=MONGODB-X509"
```

Comment:
I don't think the `<your-x509-client-subject>` part is necessary because we should read the username from the client cert automatically.

Reply:
Cool, thanks for the info. I've removed it from the uri